streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -433,6 +439,23 @@ int maybeCommitActiveTasks() {
         return active.maybeCommit();
     }
 
+    void maybePurgeCommitedRecords() {
+        // we do not check any possible exceptions since none of them are fatal
+        // that should cause the application to fail, and we will try delete with
+        // newer offsets anyways.
+        if (deleteRecordsResult == null || deleteRecordsResult.all().isDone()) {
+
+            if (deleteRecordsResult != null && deleteRecordsResult.all().isCompletedExceptionally()) {
+                log.info("Previous delete-records request has failed: {}. Try sending the new request now", deleteRecordsResult.lowWatermarks());
```

Comment:
does this need to be info level? seems more like debug to me, i.e., is a user actually going to care about this?

Reply:
Do we want to print the whole map as-is? This might be quite large. Maybe filter for failed request?